exporter/metric/metric.go

```diff
@@ -216,11 +216,22 @@ func (me *metricExporter) exportMetricDescriptor(ctx context.Context, cps export
 	// the descriptor does not exist in global cache (me.mdCache).
 	// See details in #26.
 	for kmd, md := range mds {
+		mdReq := &monitoringpb.GetMetricDescriptorRequest{
```

Comment:
Could you extract this code into a separate `getOrCreateMetricDescriptor` function to make it easier to maintain and test?

Also, do we need to cover this change with a test, or is it already covered by a mock E2E test?

Reply:
Updated.  It will get _and_ create the metric descriptor, so I named it createMetricExporterIfNeeded

It is technically covered by the mock E2E test, but the mock doesn't seem to do anything with the requests.